src/pages/patientView/PatientViewPage.tsx

```diff
@@ -93,6 +102,47 @@ export default class PatientViewPage extends React.Component<IPatientViewPagePro
         this.mutationGeneticProfileId = `${this.studyId}_mutations`;
     }
 
+    fetchHotspotsData(mutations:Mutation[]) {
+        const generateMap = function(hotspots:HotspotMutation[]) {
+            // key => geneSymbol_proteinPosition
+            // protienPosition => start[_end]
+            const map: {[key:string]: boolean} = {};
+
+            // create a map for a faster lookup
+            _.each(hotspots, function(hotspot:HotspotMutation) {
+                const positions = hotspot.residue.match(/[0-9]+/g) || []; // start (and optionally end) positions
+                const key = [hotspot.hugoSymbol.toUpperCase()].concat(positions).join("_");
```

Comment:
I was assuming the match function should return the array in the correct order, but haven't really checked the documentation.